api/src/services/files.ts

```diff
@@ -47,19 +47,23 @@ export class FilesService extends ItemsService {
 	): Promise<PrimaryKey> {
 		const storage = await getStorage();
 
-		let existingFile: Record<string, any> | null = null;
+		let existingFile: File | null = null;
```

Comment:
Lets `Pick<File, 'x'>` here to extract the fields we're reading from the DB to make it extra accurate 

```suggestion
		let existingFile: Pick<File, 'folder' | 'filename_download' | 'filename_disk' | 'title' | 'description' | 'metadata' | 'version'> | null = null;
```